### PR TITLE
CNDIT-396 : Implement xml part 1

### PR DESCRIPTION
**Pull request by raddanki-eq**
CNDIT-396 : Implement xml part 1

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -1,2 +1,3 @@
 rootProject.name = 'NEDDS-DataIngestion'
-include 'report-serive'
\ No newline at end of file
+include 'hl7-parser'
```
Comment (sdeusch) [inline settings.gradle:2]: +1

File: report-service/src/main/java/gov/cdc/dataingestion/nbs/services/NbsRepositoryServiceProvider.java [hunk 2]
```diff
@@ -51,11 +55,19 @@ public boolean saveXmlMessage(String msgXml) {
...
-    	
-    	nbsInterfaceRepo.save(item);
 
+    	nbsInterfaceRepo.save(item);
 		log.info("Persisted xml to nbs database");
 
     	return true;
     }
+
+	private long getGmtTimestamp() {
```
Comment (sdeusch) [inline report-service/src/main/java/gov/cdc/dataingestion/nbs/services/NbsRepositoryServiceProvider.java:65]: Why GMT zoned time? Where is this coming from?

UTC is the same but more common these days.

File: report-service/src/main/java/gov/cdc/dataingestion/nbs/services/NbsRepositoryServiceProvider.java [hunk 3]
```diff
@@ -29,15 +32,16 @@ public class NbsRepositoryServiceProvider {
     @Autowired
     private NbsInterfaceRepository nbsInterfaceRepo;
     
-    public boolean saveXmlMessage(String msgXml) {
+    public boolean saveXmlMessage(String xmlMsg) {
 		NbsInterfaceModel item = new NbsInterfaceModel();
 
-		item.setPayload(msgXml);
+		log.info("Xml being persisted to NBS Legacy database");
```
Comment (sdeusch) [inline report-service/src/main/java/gov/cdc/dataingestion/nbs/services/NbsRepositoryServiceProvider.java:38]: should be log.debug with a more specific message, at least an ID.

File: report-service/src/main/java/gov/cdc/dataingestion/nbs/converters/Hl7ToRhapsodysXmlConverter.java [hunk 4]
```diff
@@ -0,0 +1,1531 @@
+package gov.cdc.dataingestion.nbs.converters;
+
+import  gov.cdc.dataingestion.hl7.helper.model.hl7.group.patient.visit.PatientVisit;
+import  gov.cdc.dataingestion.hl7.helper.model.hl7.group.patient.visit.PatientVisitAdditional;
+import  gov.cdc.dataingestion.hl7.helper.model.hl7.group.order.timingQty.TimingQuantity;
+import  gov.cdc.dataingestion.hl7.helper.model.hl7.group.order.timingQty.TimingQuantityRelationship;
+import  gov.cdc.dataingestion.hl7.helper.model.hl7.messageDataType.*;
```
Comment (sdeusch) [inline report-service/src/main/java/gov/cdc/dataingestion/nbs/converters/Hl7ToRhapsodysXmlConverter.java:7]: no wildcard imports

File: report-service/src/main/java/gov/cdc/dataingestion/kafka/integration/service/KafkaConsumerService.java [hunk 5]
```diff
@@ -118,12 +120,13 @@ private void xmlConversionHandler(String message) throws Exception {
         log.info("Received message id will be retrieved from db and associated hl7 will be converted to xml");
 
         Optional<ValidatedELRModel> validatedElrResponse = this.iValidatedELRRepository.findById(message);
-        String hl7AsXml = Hl7ToXmlConverter.getInstance().convertXl7ToXml(validatedElrResponse.get().getRawMessage());
+        String hl7Msg = validatedElrResponse.get().getRawMessage();
```
Comment (sdeusch) [inline report-service/src/main/java/gov/cdc/dataingestion/kafka/integration/service/KafkaConsumerService.java:123]: Optional to make it null safe

File: report-service/build.gradle [hunk 6]
```diff
@@ -38,7 +41,30 @@ jar {
...
+				package: 'gov.cdc.dataingestion.nbs.jaxb',
+				schema: 'src/main/resources/xsd/PHDC.xsd'
+		)
+	}
+}
+
+compileJava.dependsOn jaxb
+
 dependencies {
+	implementation files('../hl7-parser/build/libs/hl7-parser.jar')
```
Comment (sdeusch) [inline report-service/build.gradle:66]: Duc already merged dependency resolution on hl7-parser into main branch. 

I  would prefer to single line like 

implementation 'gov.cdc.dataingestion:hl7-parser:0.0.1-SNAPSHOT'


if possible.

**Review by sdeusch — COMMENTED** [review]
I think the XML library should be in its own sub directory outside report-service, similar to 
hl7-parser/

**Comment by ndduc01** [comment]
Merge conflict, please resolve.

**Review by sdeusch — APPROVED** [review]

**Pull request merged**